src/core/lib/iomgr/ev_epoll_linux.c

```diff
@@ -1060,101 +1104,182 @@ static void fd_orphan(grpc_exec_ctx *exec_ctx, grpc_fd *fd,
   GRPC_ERROR_UNREF(error);
 }
 
-static grpc_error *fd_shutdown_error(grpc_fd *fd) {
-  if (!fd->shutdown) {
-    return GRPC_ERROR_NONE;
-  } else {
-    return GRPC_ERROR_CREATE_REFERENCING("FD shutdown", &fd->shutdown_error, 1);
+static void notify_on(grpc_exec_ctx *exec_ctx, grpc_fd *fd, gpr_atm *state,
+                      grpc_closure *closure) {
+  while (true) {
+    /* Fast-path: CLOSURE_NOT_READY -> <closure> */
+    /* Also do a release-cas here so that any acqire-loads in set_ready or
+       set_shutdown see this */
```

Comment:
"The release here matches acquires on load in set_ready/set_shutdown to ensure that the closure being run happens-after the call to notify_on."

This isn't a question of "see" -- the issue is what is implied by seeing.